optimum/onnxruntime/io_binding/io_binding_helper.py

```diff
@@ -0,0 +1,173 @@
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+import traceback
+from typing import Dict
+
+import numpy as np
+import torch
+
+import onnxruntime as ort
+from onnxruntime.capi.onnxruntime_inference_collection import OrtValue
+from onnxruntime.transformers.io_binding_helper import TypeHelper as ORTTypeHelper
+
+from ..utils import is_onnxruntime_training_available
+
+
+# Adapted from https://github.com/microsoft/onnxruntime/blob/93e0a151177ad8222c2c95f814342bfa27f0a64d/onnxruntime/python/tools/transformers/io_binding_helper.py#L12
+class TypeHelper(ORTTypeHelper):
+    """
+    Gets data type information of the ONNX Runtime inference session and provides the mapping from
+    `OrtValue` data types to the data types of other frameworks (NumPy, PyTorch, etc).
+    """
+
+    # TODO: Current DLPack doesn't support boolean tensor, use uint8 as workaround, remove after it is supported.
+    @staticmethod
+    def ort_type_to_numpy_type(ort_type: str):
+        ort_type_to_numpy_type_map = {
+            "tensor(int64)": np.int64,
+            "tensor(int32)": np.int32,
+            "tensor(int8)": np.int8,
+            "tensor(float)": np.float32,
+            "tensor(float16)": np.float16,
+            "tensor(bool)": np.uint8,
+        }
+        if ort_type not in ort_type_to_numpy_type_map:
+            raise ValueError(f"{ort_type} not found in map")
```

Comment:
We should improve the error message `map` might not be very helpful. 

Also, nit, but feel free to keep it.
I like it more if you check if the values are in the map and then return it rather than checking for an error.